plugins/JsTrackerInstallCheck/API.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\JsTrackerInstallCheck;
+
+use Piwik\Piwik;
+
+class API extends \Piwik\Plugin\API
+{
+    /**
+     * @var JsTrackerInstallCheck
+     */
+    protected $jsTrackerInstallCheck;
+
+    public function __construct(JsTrackerInstallCheck $jsTrackerInstallCheck)
+    {
+        $this->jsTrackerInstallCheck = $jsTrackerInstallCheck;
+    }
+
+    /**
+     * Check whether a test request has been recorded for the provided nonce
+     *
+     * @param string $idSite
+     * @param string $nonce
+     * @return array list of containers ['isSuccess' => true]
+     * @throws \Exception If the user doesn't have the right permissions
+     */
+    public function checkForJsTrackerInstallTestSuccess(string $idSite, string $nonce): array
```

Comment:
Unless your API isn't meant to support requests with multiple idsites or the magic work `all`, it might be better to define it as `int $idSite`, that way all API requests with invalid values will be prevented.
Same applies for the other methods.